### PR TITLE
open-interpreter: init at 0.1.2

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

doesn't not work on darwin because of a wrongly commited .DS_Store file, I'm trying to get upstream to remove it.
(removing it in the postpatch phase does not work unfortunately).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 254431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.litellm</li>
    <li>python310Packages.litellm.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
alright, failure is the same on linux. It appears to be because nix is trying to install this package twice with different versions
```
4.drv'
litellm> Sourcing python-remove-tests-dir-hook
litellm> Sourcing python-catch-conflicts-hook.sh
litellm> Sourcing python-remove-bin-bytecode-hook.sh
litellm> Sourcing pypa-build-hook
litellm> Using pypaBuildPhase
litellm> Sourcing pypa-install-hook
litellm> Using pypaInstallPhase
litellm> Sourcing python-imports-check-hook.sh
litellm> Using pythonImportsCheckPhase
litellm> Sourcing python-namespaces-hook
litellm> Sourcing python-catch-conflicts-hook.sh
litellm> unpacking sources
litellm> unpacking source archive /nix/store/swjxpnnky2fcnjdk8nwzfvb6zni8v7kg-source
litellm> source root is source
litellm> setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/pyproject.toml
litellm> patching sources
litellm> configuring
litellm> no configure script, doing nothing
litellm> building
litellm> Executing pypaBuildPhase
litellm> Creating a wheel...
litellm> * Getting build dependencies for wheel...
litellm> * Building wheel...
litellm> Successfully built litellm-0.1.584-py3-none-any.whl
litellm> Finished creating a wheel...
litellm> Finished executing pypaBuildPhase
litellm> installing
litellm> Executing pypaInstallPhase
litellm> Successfully installed litellm-0.1.401-py3-none-any.whl
litellm> Traceback (most recent call last):
litellm>   File "/nix/store/m4kmky8ay96qwyakxjmj1idcvrdmvb9k-python3-3.10.12/lib/python3
.10/runpy.py", line 196, in _run_module_as_main
litellm>     return _run_code(code, main_globals, None,
litellm>   File "/nix/store/m4kmky8ay96qwyakxjmj1idcvrdmvb9k-python3-3.10.12/lib/python3
.10/runpy.py", line 86, in _run_code
litellm>     exec(code, run_globals)
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/
litellm>     _main(sys.argv[1:], "python -m installer")
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/__main__.py", line 94, in _main
litellm>     installer.install(source, destination, {})
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/_core.py", line 109, in install
litellm>     record = destination.write_file(
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/destinations.py", line 207, in write_file
litellm>     return self.write_to_fs(scheme, path_, stream, is_executable)
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/destinations.py", line 167, in write_to_fs
litellm>     raise FileExistsError(message)
litellm> FileExistsError: File already exists: /nix/store/gq7a86sg8n10y2miqd23vcb8bb7agg5i-python3.10-litellm-0.1.584/lib/python3.10/site-packages/litellm/.DS_Store
```
not sure why.

**Comment by happysalada** [comment]
The dist directory was committed and was causing errors. Just tested the build on x86_64-darwin and it works.

**Comment by happysalada** [comment]
I've packaged tokentrim and open-interpreter now as well, to make it into a single PR.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 254431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>open-interpreter</li>
    <li>open-interpreter.dist</li>
    <li>python310Packages.litellm</li>
    <li>python310Packages.litellm.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.tokentrim</li>
    <li>python310Packages.tokentrim.dist</li>
    <li>python311Packages.tokentrim</li>
    <li>python311Packages.tokentrim.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 254431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>open-interpreter</li>
    <li>open-interpreter.dist</li>
    <li>python310Packages.litellm</li>
    <li>python310Packages.litellm.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.tokentrim</li>
    <li>python310Packages.tokentrim.dist</li>
    <li>python311Packages.tokentrim</li>
    <li>python311Packages.tokentrim.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 254431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>open-interpreter</li>
    <li>open-interpreter.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.litellm</li>
    <li>python310Packages.litellm.dist</li>
    <li>python310Packages.tokentrim</li>
    <li>python310Packages.tokentrim.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
    <li>python311Packages.tokentrim</li>
    <li>python311Packages.tokentrim.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 254431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>open-interpreter</li>
    <li>open-interpreter.dist</li>
    <li>python310Packages.litellm</li>
    <li>python310Packages.litellm.dist</li>
    <li>python310Packages.tokentrim</li>
    <li>python310Packages.tokentrim.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
    <li>python311Packages.tokentrim</li>
    <li>python311Packages.tokentrim.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/tokentrim/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    tiktoken
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/tokentrim/default.nix:27]: Can we enable tests?
Does this library also connect to openai?

File: pkgs/development/python-modules/tokentrim/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  pythonImportsCheck = [ "tokentrim" ];
+
+  meta = with lib; {
+    description = "Easily trim 'messages' arrays for use with GPTs";
+    homepage = "https://github.com/KillianLucas/tokentrim";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (natsukium) [inline pkgs/development/python-modules/tokentrim/default.nix:34]: Please add yourself.

File: pkgs/development/python-modules/tokentrim/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/development/python-modules/tokentrim/default.nix:27]: indeed https://github.com/KillianLucas/tokentrim/blob/main/tests/test_tokentrim.py

**Comment by natsukium** [comment]
Does `open-interpreter` work as expected?
It seems to crash in my environment.
スクリーンショット 2023-09-11 214515

**Comment by happysalada** [comment]
I've opened an issue upstream, I'm going to put this PR in draft for now, and will take it out of draft once I can confirm the issue is resolved.
https://github.com/KillianLucas/open-interpreter/issues/283

**Comment by happysalada** [comment]
@natsukium this was a bug in litellm, updating to the latest release, fixed the problem.
let me know if this is still problematic for you.

**Comment by happysalada** [comment]
(the readme doesn't say, but you need the OPENAI_API_KEY env var to be set with your openai key for this to work).

**Review by natsukium — APPROVED** [review]
It works fine on my machines.
LGTM, thanks!

**Pull request merged**